presidio-analyzer/presidio_analyzer/recognizer_registry/recognizer_registry_provider.py

```diff
@@ -0,0 +1,333 @@
+        :param registry_configuration: The configuration to use.
+        """
+
+        if conf_file and registry_configuration:
+            raise ValueError(
+                "Either conf_file or registry_configuration should"
+                " be provided, not both."
+            )
+
+        configuration = {}
+
+        if registry_configuration:
+            configuration = registry_configuration.copy()
+
+        if not conf_file:
+            configuration = self._add_missing_keys(
+                configuration=configuration, conf_file=self._get_full_conf_path()
+            )
+        else:
+            try:
+                configuration = self._add_missing_keys(
+                    configuration=configuration, conf_file=conf_file
+                )
+            except IOError:
+                logger.warning(
+                    f"configuration file {conf_file} not found.  "
+                    f"Using default config."
+                )
+                configuration = self._add_missing_keys(
+                    configuration=configuration, conf_file=self._get_full_conf_path()
+                )
+            except Exception:
+                logger.warning(
+                    f"Failed to parse file {conf_file}, " f"resorting to default"
+                )
+                configuration = self._add_missing_keys(
+                    configuration=configuration, conf_file=self._get_full_conf_path()
+                )
+
+        return configuration
+
+    @staticmethod
+    def _is_recognizer_enabled(recognizer_conf: Union[Dict[str, Any], str]) -> bool:
+        return "enabled" not in recognizer_conf or recognizer_conf["enabled"]
+
+    @staticmethod
+    def _get_recognizer_name(recognizer_conf: Union[Dict[str, Any], str]) -> str:
+        if isinstance(recognizer_conf, str):
+            return recognizer_conf
+        return recognizer_conf["name"]
+
+    @staticmethod
+    def _get_recognizer_context(
+        recognizer: Union[Dict[str, Any], str]
+    ) -> Optional[List[str]]:
+        if isinstance(recognizer, str):
+            return None
+        return recognizer.get("context", None)
+
+    @staticmethod
+    def _get_recognizer_items(
+        recognizer_conf: Union[Dict[str, Any], str]
+    ) -> dict[Any, Any] | ItemsView[str, Any]:
+        if isinstance(recognizer_conf, str):
+            return {}
+        return recognizer_conf.items()
+
+    def _get_recognizer_languages(
+        self, recognizer_conf: Union[Dict[str, Any], str]
+    ) -> List[Dict[str, Any]]:
+        """
+        Get the different language properties for each recognizer.
+
+        Creating a new recognizer for each supported language.
+        If language wasn't specified, create a recognizer for each supported language.
+
+        :param recognizer_conf: The aforementioned recognizer.
+        :return: The list of recognizers in the supported languages.
+        """
+        if (
```

Comment:
Just realized that there's a bug in here: it should be filtering out all recognizers languages that are not in supported languages...